docs/ota-client-guide/modules/ROOT/pages/add-ota-functonality-existing-yocto-project.adoc

```diff
@@ -0,0 +1,68 @@
+* `OSTREE_DEPLOY_DEVICETREE` - Set this to '1' to include devicetree(s) to boot
+* `GARAGE_SIGN_AUTOVERSION` - Set this to '1' to automatically fetch the last version of the garage tools installed by the aktualizr-native. Otherwise use the fixed version specified in the recipe.
+* `INITRAMFS_IMAGE` - initramfs/initrd image that is used as a proxy while booting into OSTree deployment. Do not change this setting unless you are sure that your initramfs can serve as such a proxy.
+* `SOTA_PACKED_CREDENTIALS` - when set, your ostree commit will be pushed to a remote repo as a bitbake step. This should be the path to a zipped credentials file in https://github.com/advancedtelematic/aktualizr/blob/master/docs/credentials.adoc[the format accepted by garage-push].
+* `SOTA_DEPLOY_CREDENTIALS` - when set to '1' (default value), deploys credentials to the built image. Override it in `local.conf` to built a generic image that can be provisioned manually after the build.
+* `SOTA_CLIENT_PROV` - which provisioning method to use. Valid options are `aktualizr-shared-prov`, `aktualizr-device-prov`, and `aktualizr-device-prov-hsm`. For more information on these provisioning methods, see the https://docs.ota.here.com/client-config/client-provisioning-methods.html[OTA Connect documentation]. The default is `aktualizr-shared-prov`. This can also be set to an empty string to avoid using a provisioning recipe.
+* `SOTA_CLIENT_FEATURES` - extensions to aktualizr. The only valid options are `hsm` (to build with HSM support) and `secondary-network` (to set up a simulated 'in-vehicle' network with support for a primary node with a DHCP server and a secondary node with a DHCP client).
+* `SOTA_SECONDARY_CONFIG` - a file containing JSON configuration for secondaries. It will be installed into `/etc/sota/ecus` on the device and automatically provided to aktualizr. See link:https://github.com/advancedtelematic/aktualizr/blob/master/docs/posix-secondaries-bitbaking.adoc[here] for more details.
+* `SOTA_HARDWARE_ID` - a custom hardware ID that will be written to the aktualizr config. Defaults to MACHINE if not set.
+* `SOTA_MAIN_DTB` - base device tree to use with the kernel. Used together with FIT images. You can change it, and the device tree will also be changed after the update.
+* `SOTA_DT_OVERLAYS` - whitespace-separated list of used device tree overlays for FIT image. This list is OSTree-updateable as well.
+* `SOTA_EXTRA_CONF_FRAGS` - extra https://lxr.missinglinkelectronics.com/uboot/doc/uImage.FIT/overlay-fdt-boot.txt[configuration fragments] for FIT image.
+* `RESOURCE_xxx_pn-aktualizr` - controls maximum resource usage of the aktualizr service, when `aktualizr-resource-control` is installed on the image. See <<aktualizr service resource control>> for details.
+* `SOTA_POLLING_SEC` - sets polling interval for aktualizr to check for updates if aktualizr-polling-sec is included in the image.
```

Comment:
Again, can we avoid copying this from meta-updater? Or would you prefer that meta-updater just points to here?

Reply:
Well, it's always been copied from meta-updater. Instead of the "ats-docs" repo hosting the dupe, it's now in 'aktualizr'.  However, deduping is the obviously the whole point of this exercise, but I want to do it in iterations with akualizr being the first repo.

I have this ticket: https://saeljira.it.here.com/browse/OTA-3568 to make sure that the "build related" (i.e. meta-updater) content is sourced directly from meta-updater rather than copying it. I will try to get it done next week.

This week's iteration is just for moving what we had in the "ats-docs" repo and getting a 1st version of the docs portal.

Reply:
Sounds fine, I understand it's gotta come one thing at a time. I kind of wish we could put it all in one repo (aktualizr, presumably), but because of differences in meta-updater/Yocto versions, I'm not sure that's actually feasible. But to make sure we're on the same page: eventually, the source of truth for the board support bits should be in meta-updater, and aktualizr should just reference/link to that?

Reply:
yes, kind of....the content will be resolved and pulled together at build time. 

The source files will contain _include_ directives that retrieve content from the meta-updater repo (however, includes unfortunately don't work when viewing *.adoc files in GitHub)

At build time, the includes will be retrieved and rendered in the HTML content.

Reply:
> however, includes unfortunately don't work when viewing *.adoc files in GitHub

Yeah, that's annoying. Is it possible to just `ifdef` them to appear as links on github?

Reply:
Yes, good idea, I can try that.